src/server/clientiface.cpp

```diff
@@ -435,32 +418,33 @@ void RemoteClient::GotBlock(v3s16 p)
 
 void RemoteClient::SentBlock(v3s16 p)
 {
-	if (m_blocks_sending.find(p) == m_blocks_sending.end())
-		m_blocks_sending[p] = 0.0f;
-	else
+	if (!m_blocks_sending.insert(p).second)
 		infostream<<"RemoteClient::SentBlock(): Sent block"
 				" already in m_blocks_sending"<<std::endl;
 }
 
-void RemoteClient::SetBlockNotSent(v3s16 p)
+void RemoteClient::SetBlockNotSent(v3s16 p, bool low_priority)
 {
-	m_nothing_to_send_pause_timer = 0;
-
 	// remove the block from sending and sent sets,
-	// and mark as modified if found
-	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0)
-		m_blocks_modified.insert(p);
+	// and reset the scan loop if found
+	if (m_blocks_sending.erase(p) + m_blocks_sent.erase(p) > 0) {
+		m_nothing_to_send_pause_timer = 0;
+		if (!m_perf_tradeoffs && !low_priority) {
+			// Reset to the maximum cube size that will
+			// guarantee that this block will the scanned again.
+			// Using m_last_center is OK, as a change in center
+			// will reset m_nearest_unsent_d to 0 anyway (set getNextBlocks).
+			p -= m_last_center;
+			m_nearest_unsent_d = std::min(m_nearest_unsent_d,
+					(s16)std::max(std::max(std::abs(p.X), std::abs(p.Y)), std::abs(p.Z)));
```

Comment:
which distance metric is this? isn't `d` an euclidean distance?
surely this would be clearer, if so:
```cpp
s16 this_d = m_last_center.getDistanceFrom(p);
m_nearest_unsent_d = std::min(m_nearest_unsent_d, this_d);
```

Reply:
yeah, it was easy enough to miss for me.